holochain_test.go

```diff
@@ -33,18 +39,19 @@ func TestNewHolochain(t *testing.T) {
 		z := Zome{Name: "zySampleZome",
 			Description: "zome desc",
 			Code:        "zome_zySampleZome.zy",
-			Entries: []EntryDef{
-				{Name: "entryTypeFoo", DataFormat: DataFormatString},
-				{Name: "entryTypeBar", DataFormat: DataFormatRawZygo},
+			Entries: map[string]EntryDef{
```

Comment:
This change to the data format is in the last commit, so we can easily split off the rest. I would also take the 'Name:' field out of the structure (have a method as needed).

As long as the array vs. hash thing is important for some other reason. I think we can either preserve order or sort by keys to get a consistent serialization.

Next I want to serialize as nested hashes, arrays and scalars so it is compatible with general serializers (I'm leaving it gob, but want it to decode to an object without go objects, just map[string] -> scalar , array, map[srting] nested. These representations are also 'native JS' as well as ruby, python, etc.